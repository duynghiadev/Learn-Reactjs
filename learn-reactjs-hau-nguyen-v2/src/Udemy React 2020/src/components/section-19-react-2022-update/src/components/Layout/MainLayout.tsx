import { Footer, Header } from '@/components/common'
import { ReactNode } from 'react'

export interface MainLayoutProps {
  children?: ReactNode
}

export function MainLayout({ children }: MainLayoutProps) {
  return (
    <div>
      <Header />
      <div>{children}</div>
      <Footer />
    </div>
  )
}
