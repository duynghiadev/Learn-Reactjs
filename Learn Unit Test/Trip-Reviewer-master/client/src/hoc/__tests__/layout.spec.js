import React from 'react';
import { create } from 'react-test-renderer';

import Layout from '../layout';

jest.mock('./../../components/Header/index', () => 'Header');
jest.mock('./../../components/Footer/index', () => 'Footer');

const mockComponent = props => <Layout {...props} />;

describe('<Layout />', () => {
  it('should render component', () => {
    const props = {
      children: <div>test</div>,
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
