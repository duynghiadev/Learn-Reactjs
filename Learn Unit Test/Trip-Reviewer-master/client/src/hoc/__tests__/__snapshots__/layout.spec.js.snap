// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render component 1`] = `
<div>
  <Header />
  <div
    style={
      Object {
        "minHeight": "95vh",
      }
    }
  >
    <div>
      test
    </div>
  </div>
  <Footer />
</div>
`;
